caddyhttp/push/handler.go

```diff
@@ -0,0 +1,70 @@
+package push
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/mholt/caddy/caddyhttp/httpserver"
+)
+
+func (h Middleware) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
+
+	pusher, hasPusher := w.(http.Pusher)
+
+	// No Pusher, no cry
+	if !hasPusher {
+		return h.Next.ServeHTTP(w, r)
+	}
+
+	// This is request for the pushed resource - it should not be recursive
+	if _, exists := r.Header[pushHeader]; exists {
+		return h.Next.ServeHTTP(w, r)
+	}
+
+	// Serve file first
```

Comment:
I didn't see that comment - if it's a rule - it should not be violated :) Fixed in #1453